modules/leaderboard/scripts/lock_board.ts

```diff
@@ -0,0 +1,52 @@
+import { ScriptContext, RuntimeError } from "../_gen/scripts/lock_board.ts";
+import { fromPrisma } from "../utils/types.ts";
+import { Leaderboard } from "../utils/types.ts";
+
+export interface Request {
+    key: string;
+}
+
+export interface Response {
+    leaderboard: Leaderboard;
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+    const { leaderboard } = await ctx.db.$transaction(async (db) => {
+        const existingLeaderboard = await db.leaderboard.findFirst({
+            where: {
+                key: req.key,
+                removedAt: null,
+            }
+        });
+
+        if (!existingLeaderboard) {
+            throw new RuntimeError(
+                "leaderboard_not_found",
+                { cause: `Leaderboard with key ${req.key} not found` },
+            );
+        }
+
+        if (existingLeaderboard.locked) {
+            throw new RuntimeError(
+                "leaderboard_locked",
+                { cause: `Leaderboard with key ${req.key} is locked` },
+            );
+        }
+
+        const newBoard = await db.leaderboard.update({
+            where: {
+                key: req.key,
+            },
+            data: {
+                locked: true,
+            }
+        });
+
+        return { leaderboard: newBoard };
```

Comment:
Is there a reason for returning an object and then destructuring it after the transaction?

Reply:
I believe it was just to make styling more consistent between transactions that returned multiple pieces of data and one piece of data

Reply:
Either way, I'm removing it